oauth2/src/main/java/com/okta/spring/boot/oauth/OktaOAuth2ResourceServerAutoConfig.java

```diff
@@ -69,13 +71,15 @@ public JwtAuthenticationConverter jwtAuthenticationConverter(OktaOAuth2Propertie
 
     @Bean
     @ConditionalOnMissingBean
+    @ConditionalOnProperty("spring.security.oauth2.client.provider.okta.issuer-uri")
     JwtDecoder jwtDecoder(OAuth2ResourceServerProperties oAuth2ResourceServerProperties,
-                          OktaOAuth2Properties oktaOAuth2Properties) {
+                          OktaOAuth2Properties oktaOAuth2Properties, Environment environment) {
 
         NimbusJwtDecoder.JwkSetUriJwtDecoderBuilder builder = NimbusJwtDecoder.withJwkSetUri(oAuth2ResourceServerProperties.getJwt().getJwkSetUri());
         builder.restOperations(restTemplate(oktaOAuth2Properties));
         NimbusJwtDecoder decoder = builder.build();
-        decoder.setJwtValidator(TokenUtil.jwtValidator(oktaOAuth2Properties.getIssuer(), oktaOAuth2Properties.getAudience()));
+        String issuerUri = environment.getProperty("spring.security.oauth2.client.provider.okta.issuer-uri");
```

Comment:
As I mentioned before, you _shouldn't_ need to inject this property, it _should_ be available from the `OAuth2ResourceServerProperties` bean. This method mixes the OAuth client and Resource server concerns.  Is that intentional?

Reply:
Got it